File: 2_avoha/backend/src/lib/notify.ts
import { env } from "../env.js";

/**
 * Discord 운영 채널 웹훅. 미설정 시 noop.
 * 네트워크 실패해도 throw 하지 않음 — 운영 사이드 채널용이라 주 플로우 차단 금지.
 */
export async function notifyOps(message: string): Promise<void> {
  if (!env.DISCORD_OPS_WEBHOOK) return;
  try {
    await fetch(env.DISCORD_OPS_WEBHOOK, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ content: message.slice(0, 1900) }),
    });
  } catch {
    // 조용히 흘려 보냄
  }
}

File: 2_Ulog/backend/app/services/notify.py
from __future__ import annotations

import httpx

from app.config import settings


async def notify_ops(message: str) -> None:
    """Discord 운영 웹훅. 미설정이면 noop. 실패해도 조용히 흘림."""
    if not settings.DISCORD_OPS_WEBHOOK:
        return
    try:
        async with httpx.AsyncClient(timeout=5.0) as client:
            await client.post(
                settings.DISCORD_OPS_WEBHOOK,
                json={"content": message[:1900]},
            )
    except Exception:
        pass
